health_rec/services/rag.py

```diff
@@ -21,147 +21,47 @@
-        )
-        if query.radius:
-            logger.info(f"Filtering services by radius: {query.radius}")
-            service_documents = [
-                doc for doc in service_documents if doc.distance <= query.radius
-            ]
-            for service in service_documents:
-                logger.info(
-                    f"Service name: {service.metadata['PublicName']}, distance: {service.distance}"
+        """Filter services by location and radius."""
+        filtered_services = []
+        for service in services:
+            if service.metadata.get("Latitude") and service.metadata.get("Longitude"):
+                service_location = (
+                    float(service.metadata["Latitude"]),
+                    float(service.metadata["Longitude"]),
                 )
-        return list(service_documents)
-
-    def _prepare_context(
-        self, service_documents: List[ServiceDocument]
-    ) -> Tuple[str, bool]:
-        """Prepare context from service documents."""
-        context = "\n".join([service.document for service in service_documents])
-        no_services_found = False
-        if not context:
-            context = "No services found within the specified radius."
-            no_services_found = True
-        return context, no_services_found
-
-    def _convert_documents_to_services(
-        self, service_documents: List[ServiceDocument]
-    ) -> List[Service]:
-        """Convert ServiceDocument objects to Service objects."""
-        return [_metadata_to_service(doc.metadata) for doc in service_documents]
+                distance = _calculate_distance(service_location, location)
+                if distance <= radius:
+                    service.distance = distance
+                    filtered_services.append(service)
+        return filtered_services
 
     def _generate_response(self, query: str, context: str) -> RecommendationResponse:
-        """
-        Generate a response based on the input query and context using RAG methodology.
-
-        Parameters
-        ----------
-        query : str
-            The user's input query for which a recommendation is requested.
-        context : str
-            The context of relevant services.
-
-        Returns
-        -------
-        RecommendationResponse
-            An object containing the generated recommendation and relevant services.
-        """
+        """Generate a response using the RAG prompt."""
         generation_template = """
-        You are an expert with deep knowledge of health and community services. You will be providing a recommendation to an individual who is seeking help. The individual is seeking help with the following query:
+        You are an expert with deep knowledge of health and community services in the Greater Toronto Area (GTA). You will be providing a recommendation to an individual who is seeking help. The individual is seeking help with the following query:
```

Comment:
That makes sense, I'll make it general again since the dataset is more than just GTA and I'll add a note in the eval documentation regarding the change of prompt, shouldn't affect the retrieval benchmarks though.